pulsar/consumer_partition.go

```diff
@@ -499,6 +499,12 @@ func (pc *partitionConsumer) ackID(msgID MessageID, withResponse bool) error {
 	trackingID := toTrackingMessageID(msgID)
 
 	if trackingID != nil && trackingID.ack() {
+		trackingID = &trackingMessageID{
```

Comment:
It is best to add some comments explaining why, which can be intuitively understood by developers who are not familiar with this implementation

Reply:
Good suggestion. I agree with you.

Reply:
+1

Reply:
Addressed.